src/backend/logger/__tests__/logfile.test.ts

```diff
@@ -0,0 +1,186 @@
+        )
+      ],
+      { prefix: 'Backend', skipLogToFile: true }
+    )
+  })
+
+  test('createNewLogFileAndClearOldOnces success', () => {
+    jest.spyOn(app, 'getPath').mockReturnValue(tmpDir.name)
+    jest.spyOn(configStore, 'has').mockReturnValue(true)
+    jest.spyOn(configStore, 'get').mockReturnValue({
+      currentLogFile: 'old/log/path/file.log',
+      lastLogFile: undefined
+    })
+
+    const data = logfile.createNewLogFileAndClearOldOnces()
+
+    expect(logError).not.toBeCalled()
+    expect(data).toStrictEqual({
+      currentLogFile: expect.any(String),
+      lastLogFile: 'old/log/path/file.log'
+    })
+  })
+
+  test('createNewLogFileAndClearOldOnces removing old logs fails', () => {
+    jest.spyOn(app, 'getPath').mockReturnValue(tmpDir.name)
+    const spyUnlinkSync = jest
+      .spyOn(graceful_fs, 'unlinkSync')
+      .mockImplementation(() => {
+        throw Error('unlink failed')
+      })
+    const date = new Date()
+    date.setMonth(date.getMonth() > 0 ? date.getMonth() - 1 : 11)
+    const monthOutdated = join(
+      tmpDir.name,
+      // @ts-ignore replaceAll error
+      `heroic-${date.toISOString().replaceAll(':', '_')}.log`
+    )
+
+    graceful_fs.openSync(monthOutdated, 'w')
+
+    expect(graceful_fs.existsSync(monthOutdated)).toBeTruthy()
+
+    const data = logfile.createNewLogFileAndClearOldOnces()
+
+    expect(logError).toBeCalledWith(
+      [
+        expect.stringContaining('Removing old logs in /tmp/'),
+        Error('unlink failed')
+      ],
+      { prefix: 'Backend', skipLogToFile: true }
+    )
+    expect(graceful_fs.existsSync(monthOutdated)).toBeTruthy()
+  })
+
+  test('createNewLogFileAndClearOldOnces removing old logs successful', () => {
+    jest.spyOn(app, 'getPath').mockReturnValue(tmpDir.name)
+    const date = new Date()
+    date.setMonth(date.getMonth() > 0 ? date.getMonth() - 1 : 11)
+    const monthOutdated = join(
+      tmpDir.name,
+      // @ts-ignore replaceAll error
+      `heroic-${date.toISOString().replaceAll(':', '_')}.log`
+    )
+    date.setFullYear(2021)
+    const yearOutdated = join(
+      tmpDir.name,
+      // @ts-ignore replaceAll error
+      `heroic-${date.toISOString().replaceAll(':', '_')}.log`
+    )
+
+    graceful_fs.openSync(monthOutdated, 'w')
+    graceful_fs.openSync(yearOutdated, 'w')
+
+    expect(graceful_fs.existsSync(monthOutdated)).toBeTruthy()
+    expect(graceful_fs.existsSync(yearOutdated)).toBeTruthy()
+
+    const data = logfile.createNewLogFileAndClearOldOnces()
+
+    expect(logError).not.toBeCalled()
+    expect(graceful_fs.existsSync(monthOutdated)).toBeFalsy()
+    expect(graceful_fs.existsSync(yearOutdated)).toBeFalsy()
+  })
+
+  test('getLogFile all possible values', () => {
+    //@ts-ignore Needed override the currentLogFile
+    constants.currentLogFile = 'current.log'
+    //@ts-ignore Needed override the lastLogFile
+    constants.lastLogFile = 'last.log'
```

Comment:
I can mock the values globally, but somehow i can't override them in the test. This was the easiest way. I can try again.

Reply:
> There is no other easy way i found. Feel free to find one.

Not sure what you mean, the code snippet in my original comment works (without giving me warnings/having to use `@ts-ignore`)

Reply:
It removes the type error but has the same problem. You still need to override the constant back to the default one.
I want to find a better way.

Reply:
I have some ideas on how to solve this with Vitest, so it's fine for now

Reply:
I changed it to your solution, the problem is the restoring of the default value. 
Please keep the global manual mocks in vitest, because they will pretty helpful if we introduce more tests.

Reply:
Wasn't planning on removing 'em